lib/FrameManager.js

```diff
@@ -719,6 +722,20 @@ class Frame {
     return new WaitTask(this, pageFunction, 'function', polling, timeout, ...args).promise;
   }
 
+  async waitForNavigation(options = {}) {
+    const timeout = typeof options.timeout === 'number' ? options.timeout : 30000;
+    const NavigatorWatcher = require('./NavigatorWatcher');
```

Comment:
Can you move this require up top with the others? Keeps things organized.

Reply:
This will require some refactoring, because inside `NavigatorWatcher`, we require `FrameManager`. So one of them is going to be undefined inside the other if we move this require to the top.
I'm open for suggestions.